assets/js/search-bar.js

```diff
@@ -137,6 +143,66 @@ function addEventListeners () {
     clearSearch()
     hideAutocomplete()
   })
+
+  window.addEventListener('online', enableRemoteSearch)
+  window.addEventListener('offline', disableRemoteSearch)
+}
+
+function enableRemoteSearch (_event) {
+  Array.from(document.getElementsByClassName('online-only')).forEach(element => {
+    element.removeAttribute('disabled')
+    element.removeAttribute('title')
+    updateSearchTypeOptions(element)
+  })
+}
+
+function disableRemoteSearch (_event) {
+  Array.from(document.getElementsByClassName('online-only')).forEach(element => {
+    element.setAttribute('disabled', '')
+    element.setAttribute('title', 'Local searching only - browser is offline')
+    updateSearchTypeOptions(element)
+  })
+}
+
+function updateSearchTypeOptions (element) {
+  Array.from(element.getElementsByTagName('option')).forEach(option => {
+    if (option.value === 'related') {
+      if (shouldEnableRelatedSearch()) {
+        option.removeAttribute('disabled')
+        option.removeAttribute('title')
+      } else {
+        option.setAttribute('disabled', '')
+        option.setAttribute('title', 'No known related packages to search')
+      }
+    }
+    if (option.value === 'latest') {
+      if (shouldEnableLatestSearch()) {
+        option.removeAttribute('disabled')
+        option.removeAttribute('title')
+      } else {
+        option.setAttribute('disabled', '')
+        option.setAttribute('title', 'Already browsing latest version')
+      }
+    }
+  })
+}
+
+function shouldEnableRelatedSearch () {
+  return getSearchNodes().length > 1
+}
+
+function shouldEnableLatestSearch () {
+  const versionNodes = getVersionNodes()
+
+  if (versionNodes.length > 0) {
+    const latest = versionNodes[0]
+    const searchNodes = getSearchNodes()
+    const match = searchNodes.some(node => `v${node.version}` === latest.version)
```

Comment:
I don't think we need to check if latest or not. It is easier if the user doesn't have to think about it. Regardless if it we are in the latest package or not, we should allow latest to be chosen (unless there is no searchNodes). So basically, we will have this:

1. If searchNodes.length > 1, the related option is enabled, otherwise it is disabled
2. If searchNodes.length > 0, the latest option is enabled, otherwise it is disabled
3. Otherwise it is only current/local

If online, the default should be related > latest > current. If offline, it is the current one.

In other words, this should return `return versionNodes.length > 0`.